cmake/toolchain/llvm/generic.cmake

```diff
@@ -11,13 +11,13 @@ endif()
 set(LLVM_TOOLCHAIN_PATH ${CLANG_ROOT_DIR} CACHE PATH "clang install directory")
 
 set(COMPILER clang)
-if(CONFIG_LLVM_USE_LD)
-  set(LINKER ld)
-else()
-  set(LINKER lld)
-endif()
 set(BINTOOLS llvm)
 
+# Note that ${LINKER} is dependent on CONFIG_LLVM_USE_LLD
+# or CONFIG_LLVM_USE_LD, and these kconfigs are not
+# available when this file is parsed. So setting ${LINKER}
+# needs to be deferred to target.cmake.
+
```

Comment:
I put it here because it deviates from other toolchains' `generic.cmake`, simply trying to avoid confusing future readers.

Reply:
> I put it here because it deviates from other toolchains' generic.cmake, simply trying to avoid confusing future readers.

I fully understand why you did it, but I believe it would then be better to make a general comment in the top of this file describing the purpose of the file.

tbhI also think we should cleanup the other files, so that generic.cmake and target.cmake contains what they really should contain and not messed like now. But I consider that out-of-scope for this PR.